File: Clients/BlogBlazor/Shared/Blazor.Contracts/V1/Validators/CommentsValidators/CreateCommentRequestValidator.cs
using Blazor.Contracts.V1.Requests.CommentsRequests;
using FluentValidation;

namespace Blazor.Contracts.V1.Validators.CommentsValidators
{
    /// <summary>
    /// Create comment request validator.
    /// </summary>
    /// <seealso cref="AbstractValidator{CreateCommentRequest}" />
    public class CreateCommentRequestValidator : AbstractValidator<CreateCommentRequest>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="CreateCommentRequestValidator"/> class.
        /// </summary>
        public CreateCommentRequestValidator()
        {
            RuleFor(x => x.PostId)
                .NotEqual(-1);

            RuleFor(x => x.CommentBody)
                .NotEmpty()
                .Matches("^[a-zA-Z0-9 ]*S");

            RuleFor(x => x.UserId)
                .NotEmpty();
        }
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Validators/CommentsValidators/CreateCommentRequestValidator.fs
namespace Blog.FSharp.Contracts.V1.Validators.CommentsValidators

open FluentValidation
open Blog.FSharp.Contracts.V1.Requests.CommentsRequests

/// <summary>
/// Create comment request validator.
/// </summary>
type CreateCommentRequestValidator() =
    inherit AbstractValidator<CreateCommentRequest>()

    do
        // PostId should not be -1
        base.RuleFor(fun x -> x.PostId)
            .NotEqual(-1) |> ignore

        // CommentBody is required and shold match to Regex
        base.RuleFor(fun x -> x.CommentBody)
            .NotEmpty()
            .Matches("^[a-zA-Z0-9 ]*S") |> ignore

        // UserId is required
        base.RuleFor(fun x -> x.UserId)
            .NotEmpty() |> ignore